tests/test_cvc.ml

```diff
@@ -0,0 +1,109 @@
+open OUnit2
+
+let rsa_suite =
+  let open Key_parsers in
+  let open Asn1.RSA_CVC in
+  let expected_public =
+    let n =
+      Z.of_string
+        "0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\
+        FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\
+        FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF\
+        FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF"
+    in
+    let e = Z.of_string "0x010001" in
+    Public.{n; e}
+  in
+  let cmp = Z.equal in
+  let printer = Z.to_string in
+  let test_pub ~decode (expected : Public.t) cvc ctxt =
+    let real = decode cvc in
+    let open Public in
+    Test_util.assert_ok real @@ function
+      | `RSA real ->
+          assert_equal ~ctxt ~cmp ~printer ~msg:"n" expected.n real.n;
+          assert_equal ~ctxt ~cmp ~printer ~msg:"e" expected.e real.e
+      | `ECDSA _
+      | `UNKNOWN ->
+          assert_failure ""
+  in
+  let cvc_suite =
+    let cvc = Cstruct.of_string @@ [%blob "../tests/keys/rsa_cvc_dummy.key"] in
+    [ "Public" >:: test_pub ~decode:Asn1.CVC.decode expected_public cvc
+    ]
+  in
+  [ "CVC" >::: cvc_suite
+  ]
+
+let ecdsa_suite =
+  let open Key_parsers in
+  let open Asn1.ECDSA_CVC in
+  let expected_public =
+    let modulus =
+      Z.of_string
+        "0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF"
+    in
+    let coefficient_a =
```

Comment:
Changed, should be in the next push (assuming you meant that I should use a random/ascending value for `public_point_y` rather than `cofactor_f`).